pallets/extended-assets/src/lib.rs

```diff
@@ -341,6 +341,35 @@ pub mod pallet {
 
             Ok(())
         }
+
+        /// Marks an asset as regulated, representing that the asset will only operate between KYC-verified wallets.
+        ///
+        /// ## Parameters
+        ///
+        /// - `origin`: The origin of the transaction.
+        /// - `asset_id`: The identifier of the asset.
+        #[pallet::call_index(4)]
+        #[pallet::weight(<T as Config>::WeightInfo::regulate_asset())]
+        pub fn regulate_asset(origin: OriginFor<T>, asset_id: AssetIdOf<T>) -> DispatchResult {
```

Comment:
As I remember, the main idea is to allow to operate with the asset only for KYC passed users.
What if the asset is registered and distributed between non-KYC users and then after that we will regulate the asset?
In this case we have non-KYC users with regulated asset.

Reply:
Yes, it's possible for asset owner to regulate the asset after distribution.
Simply enough they can force the users to verify KYC by regulating the asset in order to operate further.